huaweicloud/services/ims/data_source_huaweicloud_images_images.go

```diff
@@ -0,0 +1,318 @@
+				Computed: true,
+			},
+			"updated_at": {
+				Type:     schema.TypeString,
+				Computed: true,
+			},
+			"size_bytes": {
+				Type:     schema.TypeInt,
+				Computed: true,
+			},
+		},
+	}
+}
+
+// dataSourceImagesImagesRead performs the image lookup.
+func dataSourceImagesImagesRead(ctx context.Context, d *schema.ResourceData, meta interface{}) diag.Diagnostics {
+	config := meta.(*config.Config)
+	imageClient, err := config.ImageV2Client(config.GetRegion(d))
+	if err != nil {
+		return fmtp.DiagErrorf("Error creating HuaweiCloud image client: %s", err)
+	}
+
+	imageType := d.Get("visibility").(string)
+	if imageType == "public" {
+		imageType = "gold"
+	}
+
+	listOpts := cloudimages.ListOpts{
+		Name:           d.Get("name").(string),
+		Owner:          d.Get("owner").(string),
+		SortKey:        d.Get("sort_key").(string),
+		SortDir:        d.Get("sort_direction").(string),
+		Tag:            d.Get("tag").(string),
+		Platform:       d.Get("os").(string),
+		OsVersion:      d.Get("os_version").(string),
+		Architecture:   d.Get("architecture").(string),
+		VirtualEnvType: d.Get("image_type").(string),
+		FlavorId:       d.Get("flavor_id").(string),
+		Imagetype:      imageType,
+		Status:         "active",
+	}
+
+	if epsId := common.GetEnterpriseProjectID(d, config); epsId != "" {
+		listOpts.EnterpriseProjectID = epsId
+	} else {
+		listOpts.EnterpriseProjectID = "all_granted_eps"
+	}
+
+	logp.Printf("[DEBUG] List Options: %#v", listOpts)
+
+	allPages, err := cloudimages.List(imageClient, listOpts).AllPages()
+	if err != nil {
+		return fmtp.DiagErrorf("Unable to query images: %s", err)
+	}
+
+	allImages, err := cloudimages.ExtractImages(allPages)
+	if err != nil {
+		return fmtp.DiagErrorf("Unable to retrieve images: %s", err)
+	}
+
+	if len(allImages) < 1 {
+		return fmtp.DiagErrorf("Your query returned no results. " +
+			"Please change your search criteria and try again.")
+	}
+
+	var nameRegexRes *regexp.Regexp
+	if nameRegex, ok := d.GetOk("name_regex"); ok {
+		nameRegexRes, err = regexp.Compile(nameRegex.(string))
+		if err != nil {
+			return fmtp.DiagErrorf("name_regex format error: %s", err)
+		}
+	}
+	var resultImages []interface{}
+	var ids []string
+	for _, item := range allImages {
+		image := item
+		if nameRegexRes != nil && !nameRegexRes.MatchString(image.Name) {
+			continue
+		}
+		ids = append(ids, image.ID)
+		resultImages = append(resultImages, flattenImage(&image))
+	}
+	if len(resultImages) < 1 {
+		return fmtp.DiagErrorf("Your query returned no results. " +
+			"Please change your search criteria and try again.")
+	}
+	mErr := d.Set("images", resultImages)
+	if mErr != nil {
+		return fmtp.DiagErrorf("set images err:%s", mErr)
+	}
+
+	d.SetId(hashcode.Strings(ids))
+
+	return nil
+}
+
+func flattenImage(image *cloudimages.Image) map[string]interface{} {
+	res := map[string]interface{}{
+		"id":                    image.ID,
+		"name":                  image.Name,
+		"visibility":            image.Imagetype,
```

Comment:
```suggestion
		"visibility":            image.visibility,
```
right?

Reply:
yes, this field is used now

Reply:
The visibility field was first used, but this field does not support the type of market, so we use the field Imaetype, for backward compatibility, the original fields are used to return.